couchdb/server/helpers.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+#
+from inspect import getsource
+from textwrap import dedent
+from types import FunctionType
+
+from couchdb import util
+
+
+def maybe_extract_source(fun):
+    if isinstance(fun, FunctionType):
+        return dedent(getsource(fun))
```

Comment:
Err... drop this patch please.
`dedent` is for `FunctionType`.

In `maybe_extract_source`, if the input is string, it just return the original string.